src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -309,6 +309,68 @@ private static function getExpressionResolvers(): array
 						)
 					);
 				},
+				'isInstanceOfAny' => function (Scope $scope, Arg $expr, Arg $class): ?\PhpParser\Node\Expr {
+					$classType = $scope->getType($class->value);
+					if (!$classType instanceof ConstantArrayType) {
+						return null;
+					}
+
+					$constantStringTypes = $classType->getValueTypes();
+
+					// If array is empty, it's no assert at all
+					if (count($constantStringTypes) === 0) {
+						return null;
+					}
+
+					$orRight = array_pop($constantStringTypes);
+
+					if (!$orRight instanceof ConstantStringType) {
+						throw new \PHPStan\ShouldNotHappenException();
```

Comment:
You can't throw ShouldNotHappenException here - that would lead to an `Internal error` printed to the user. You need to end the execution gracefully by returning `null`.

Reply:
Okay. Should I return null or just skip all array entries which are not of type `ConstantStringType`? That way if you provide some correct values they would still get respected.

Reply:
Well, if the arguments are `[ConstantStringType, StringType, ConstantStringType]` then the best thing we can say the first argument is `ObjectWithoutClassType`, so there should be some logic like that.

You could look at `ConstantArrayType::getItemType()` - it will contain a Type of all the arguments - so it will be a `StringType` in case of `[ConstantStringType, StringType, ConstantStringType]` and it will be a `UnionType` of `ConstantStringType|ConstantStringType|ConstantStringType` in case of `[ConstantStringType, ConstantStringType, ConstantStringType]`.

So you could do something like this:

```php
$itemType = ConstantArrayType::getItemType();
$constantStrings = TypeUtils::getConstantStrings($itemType);
if (count($constantStrings) === 0) {
    return new ObjectWithoutClassType();
}

// iterate over the constant string types
```